### ISSUE TITLE
Inconsistency in `common_time` with `'bin'`: are the bins closed or open on the right?

### ISSUE THREAD

**Issue by MarcoGorelli** [LinkedEarth/Pyleoclim_util#340]
**Describe the bug**
It's not clear what the bins are, nor how values are assigned to them.

**To Reproduce**
```python
ts1 = pyleo.Series(
    time=np.array([1, 2, 3]),
    value=np.array([8, 3, 5]),
    time_unit='yr BP',
)
ts2 = pyleo.Series(
    time=np.array([1, 2, 3]),
    value=np.array([6, 2, 1]),
    time_unit='yr BP',
)
ms = pyleo.MultipleSeries([ts1, ts2])
newms = ms.common_time('bin')
print(newms.series_list)
```

This gives
```
{'log': ({0: 'clean_ts', 'applied': True, 'verbose': False},)}
{'log': ({0: 'clean_ts', 'applied': True, 'verbose': False},)}
[None
time [yr BP]
1.5    8.0
2.5    4.0
Name: value, dtype: float64, None
time [yr BP]
1.5    6.0
2.5    1.5
Name: value, dtype: float64]
```
which I find quite odd. It looks like we have:
- `1.5` means `[1, 2)`
- `2.5` means `[2, 3]`

In the first case, the bin is closed on the left and open on the right. But in the second case, it is closed on both sides.

**Expected behavior**
I'd have expected:
- `1.5` to mean `[1, 2)`
- `2.5` to mean `[2, 3)`
- `3.5` to mean `[3, 4)`

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] linux
 - Browser [e.g. chrome, safari] irrelevant
 - Version [e.g. 22] Development branch, commit e606f8

**Comment by alexkjames** [comment]
Hi marco, currently the way bin edges are defined is all but the last bin are half open. It is a bit confusing and should be specified more clearly in the docstring. I'm not sure about the solution you propose in #341 though, wouldn't this result in data from outside of the range defined by start and stop being included in the binned series? I suppose I'm not totally clear on the advantage of this approach (which may very well be on me!).

**Comment by alexkjames** [comment]
sounds good, will do!

**Comment by MarcoGorelli** [comment]
closed by #347
